File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-apps/src/services/appsService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { AppsPageResult } from "../types/appsModels.js";

/**
 * Apps service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface AppsService {
  readonly capability: "apps";
}

export function createAppsService(client: AppstoreAppClient): AppsService {
  void client;
  return { capability: "apps" };
}

export function emptyAppsResult(): AppsPageResult<never> {
  return { items: [] };
}

File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_apps/lib/src/services/apps_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/apps_models.dart';

/// Apps service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class AppsService {
  const AppsService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'apps';

  AppsPageResult<Never> empty() =>
      const AppsPageResult<Never>(items: <Never>[]);
}
